crates/ra_hir_ty/src/db.rs

```diff
@@ -48,6 +48,12 @@ pub trait HirDatabase: DefDatabase + Upcast<dyn DefDatabase> {
     #[salsa::invoke(crate::callable_item_sig)]
     fn callable_item_signature(&self, def: CallableDef) -> PolyFnSig;
 
+    #[salsa::invoke(crate::lower::return_type_impl_traits)]
+    fn return_type_impl_traits(
```

Comment:
Let's add this to our Archimedes' method [memory usage measurement](https://github.com/rust-analyzer/rust-analyzer/blob/abf7d1747d9910e7b4e11357ae9bcf5c594f0d55/crates/ra_ide_db/src/change.rs#L302-L378)